### PR TITLE
[ci-skip][Docs]Add sqlite3_production_warning to configuring guide

### PR DESCRIPTION
**Pull request by hachi8833**
### Motivation / Background

This Pull Request has been created because the `config.active_record.sqlite3_production_warning` config that was merged as https://github.com/rails/rails/pull/42191 in [Rails 7.0.0.alpha1](https://github.com/rails/rails/blob/7-0-stable/activerecord/CHANGELOG.md#rails-700alpha1-september-15-2021) still has not been added to the configuring guide.

### Detail

This Pull Request adds the entry for `config.active_record.sqlite3_production_warning`.

### Additional information

As far as I see, the current config entries for Active Record are unsorted and I'm a bit wondering if the line I added is appropriate or not. This time I added it to the next line of `sqlite3_adapter_strict_strings_by_default`.

Or if you think adding `config.active_record.sqlite3_production_warning` to the configuring guide is redundant because the warning indicates the one, just close the PR.

```
Oct 14 05:38:10 PM  W, [2023-10-14T08:38:07.617001 #1]  WARN -- : You are running SQLite in production, this is generally not recommended. You can disable this warning by setting "config.active_record.sqlite3_production_warning=false".
```

cc: @intrip

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`

### CODE REVIEW

**Comment by zzak** [comment]
Should this be added to the default values as well?

https://github.com/rails/rails/blob/94ff1bf0dcc38cc9176844d90ecfa54a5a09dada/guides/source/configuring.md?plain=1#L92

**Comment by hachi8833** [comment]
@zzak That's what I want to know about.

Current CI makes us to add the entry to the new framework file as well if we add the entry to the list you specified. The config just control the warning and irrelevant to compatibility issuees, so I'm afraid It might be unnecessary to add. 🤔

I'd be glad if some core members would suggest about this.

**Comment by zzak** [comment]
This is just my opinion, but not a very strong one, I could be wrong.

> Below are the default values associated with each target version. In cases of conflicting values, newer versions take precedence over older versions.

It doesn't explicitly say "all default values" for each version, but to me it looks like it could also be seen as a Table of Contents organized by version, where the contents below this are sorted by framework.

File: guides/source/configuring.md [hunk 1]
```diff
@@ -1458,6 +1458,17 @@ The default value depends on the `config.load_defaults` target version:
...
+
+Specifies whether a warning should be raised when using SQLite3 in production environment.
+
+The default value depends on the `config.load_defaults` target version:
+
+| Starting with version | The default value is |
+| --------------------- | -------------------- |
+| (original)            | `false`              |
+| 7.0                   | `true`               |
+
```
Comment (skipkayhil) [inline guides/source/configuring.md:1471]:
```suggestion
The default value is `true`
```

**Review by skipkayhil — COMMENTED** [review]
The first line of "Versioned Default Values" defines "default values" for a "target version" in reference to `load_defaults` specifically:
> `config.load_defaults` loads default configuration values for a target version...
> 
> Below are the default values associated with each target version...
> #### Default Values for Target Version ...

Since this configuration is not part of `load_defaults`, it should not include the table in the description, and I don't think it should be included in the "Versioned Default Values" section either.

**Comment by zzak** [comment]
> Since this configuration is not part of load_defaults, it should not include the table in the description, and I don't think it should be included in the "Versioned Default Values" section either.

@skipkayhil Maybe we can make that clearer, instead of  "Default Values for Target Version 7.0" as each heading just use `config.load_defaults(7.0)`. Or a combination so that it's obvious looking at this list 100 lines down what it's for.

**Comment by intrip** [comment]
SQLite is becoming a more capable database and there are more companies using it in production. It is really good for self-hosted/small environments where you don't want to boot a separate server for the DB. I've noticed that there was also some recent work on improving the SQLite integration in Rails. I propose to consider the removal of this warning from Rails, doing it would also remove the need to handle Rails defaults.
/cc @dhh

**Comment by dhh** [comment]
Yes, I think we should do away with this warning. Replace it with a notice in the docs/guide explaining the limitations of SQLite, as well as the advantages.

**Comment by hachi8833** [comment]
I close this PR and will recreate as a new one for 7-0/7-1-stable branches.

**Pull request closed without merging**